### ISSUE TITLE
Animaton start

### ISSUE THREAD

**Issue by rafaelcamargo** [glorious-codes/glorious-demo#57]
To avoid users to miss the animation, play it only when its container gets visible on screen.

**Comment by rafaelcamargo** [comment]
General idea is:

``` javascript
// index.js
import { ElementPosition } from './services/element-position/element-position';
end(){
  const player = new Player(this.container, this.steps);
  const elementPosition = new ElementPosition(this.container);
  elementPosition.executeWhenAbovePageFold(player.play);
}

// element-position.js
import { generate } from 'nanoid/generate';

export class ElementPosition {
  constructor(element){
    this.element = element;
    this.timeoutId = null;
    setElementUniqueSelectorId(this.element);
  },
  executeWhenAbovePageFold(callback){
    if(!isAboveThefold(this.element))
      return this.setTimeoutId(setTimeout(() => executeWhenAbovePageFold(callback), 1000));
    return handleCallbackExecution(callback, this.timeoutId);
  },
  setTimeoutId(id){
    this.timeoutId = id;
  }
}

function setElementUniqueSelectorId(element){
  element.setAttribute('data-glorious-demo-id', generate());
}

function handleCallbackExecution(callback, timeoutId){
  clearTimeout(timeoutId);
  callback();
}
```

**Comment by rafaelcamargo** [comment]
Extract [this service](https://github.com/glorious-codes/glorious-pitsby/blob/master/src/webapp/scripts/services/page-fold.js) to an external lib - @glorious/fold, maybe - and use it in here. Doing so, *element-position* will be no longer necessary.

**Comment by heowc** [comment]
Awesome!

**Comment by heowc** [comment]
As I said before, I created a [hexo tag library](https://github.com/heowc/hexo-tag-gdemo). :smile:
With this added, I think we can make a much more static page generator. :rocket:

**Comment by rafaelcamargo** [comment]
Hello, @heowc! Improvement just released on [v0.9.0](https://github.com/glorious-codes/glorious-demo/releases/tag/v0.9.0). Thanks for using Glorious Demo 😉

**Comment by heowc** [comment]
Oh! Thank you for applying for quick features. I also upgraded to [0.11.0](https://www.npmjs.com/package/hexo-tag-gdemo/v/0.11.0).
